hail/python/hail/backend/service_backend.py

```diff
@@ -418,34 +418,40 @@ async def _rpc(self,
                     raise
 
             with timings.step("read output"):
+                result_bytes = await retry_transient_errors(self._read_output, ir, iodir + '/out')
+                return token, result_bytes, timings
+
+    async def _read_output(self, ir: Optional[BaseIR], output_uri: str) -> bytes:
+        assert self._batch
+
+        try:
+            driver_output = await self._async_fs.open(output_uri)
+        except FileNotFoundError as exc:
+            raise FatalError('Hail internal error. Please contact the Hail team and provide the following information.\n\n' + yamlx.dump({
+                'service_backend_debug_info': self.debug_info(),
+                'batch_debug_info': await self._batch.debug_info()
+            })) from exc
+
+        async with driver_output as outfile:
+            success = await read_bool(outfile)
+            if success:
+                result_bytes = await read_bytes(outfile)
                 try:
-                    driver_output = await self._async_fs.open(iodir + '/out')
-                except FileNotFoundError as exc:
+                    return result_bytes
+                except orjson.JSONDecodeError as err:
```

Comment:
Shouldn't the `read_bytes` line go in the try? Also, is the `orjson.JSONDecodeError` for if the HTTP response body is somehow malformed? How would that happen?